### DISCUSSION TITLE
Why islands scripts are pushed to opts.imports?

### DISCUSSION THREAD

**Discussion by pazguille** [freshframework/fresh#747] [Q&A] [answered]
Hi,

I would like to understand because there are performance improvements about how scripts are loaded. Currently, the script have more importance than LCP images.

The scripts download, parse and execute blocking the render. Maybe, they could be loaded as `modulepreload` using `opts.preloads` instead of `opts.imports`.

Current:

<img width="1085" alt="fresh_-_The_next-gen_web_framework_" src="https://user-images.githubusercontent.com/250856/189349728-559e9a4c-e568-4c09-958a-288f39c5d04d.png">

Removing scripts from `opts.imports`:

<img width="1129" alt="fresh_-_The_next-gen_web_framework_" src="https://user-images.githubusercontent.com/250856/189352796-754725dd-0258-46bc-ab56-97cdaedf73d8.png">

**Comment by lucacasonato** [comment]
To preload them. Not all browsers support `modulepreload` unfortunately.

**Comment by pazguille** [comment]
Ok! We could add the defer attribute on these scripts to not block the HTML parser: https://v8.dev/features/modules#defer

What do you think? 

Also add an option to use preloads instead imports.

**Reply by lucacasonato** [reply]
type=module is async by default.

**Reply by pazguille** [reply]
You are right. I missed the `type="module"` here: https://github.com/denoland/fresh/blob/main/src/server/render.ts#L346

**Comment by lucacasonato** [comment, marked as answer]
If you look at the waterfall in DevTools you should notice that the script tags are not actually blocking anything. They just show up first in the network panel because the requests are sent off first (because they are encountered by the HTML parser first).
